packages/application-components/src/components/tabular-detail-page/tabular-detail-page.tsx

```diff
@@ -0,0 +1,82 @@
+import type { ReactElement, ReactNode, MouseEvent, KeyboardEvent } from 'react';
+import Spacings from '@commercetools-uikit/spacings';
+import { sharedMessages } from '@commercetools-frontend/i18n';
+import PageHeaderTitle from '../internals/page-header-title';
+import PageTopBar from '../internals/page-top-bar';
+import {
+  ControlsContainter,
+  TabularPageContainer,
+} from '../internals/tabular-page';
+import { ContentWrapper, PageWrapper } from '../internals/page.styles';
+
+// NOTE: the `MessageDescriptor` type is exposed by `react-intl`.
+// However, we need to explicitly define this otherwise the prop-types babel plugin
+// does not recognize the object shape.
+type MessageDescriptor = {
+  id: string;
+  description?: string | object;
+  defaultMessage?: string;
+};
+
+type TTabularDetailPageProps = {
+  /**
+   * The title of the page.
+   */
+  title: string;
+  /**
+   * The subtitle of the page.
+   */
+  subtitle?: string | ReactElement;
```

Comment:
nit: maybe we could use `ReactNode` instead as we did in the ui-kit.

Reply:
Thanks. The idea was to keep the API of all page components coherent. Would you be ok with leaving the type as it is?

Reply:
Both forms are valid and the one proposed its more compact but one could argue that it could be too flexible (I don't see it that way).
I proposed it because we changed this same use case in many PRs form the ui-kit migration to Typescript.

Anyway, if we want to keep it like it is now, it's ok from my side.